chemprop/cli/hpopt.py

```diff
@@ -225,7 +225,7 @@ def process_hpopt_args(args: Namespace) -> Namespace:
 
 
 def build_search_space(search_parameters: list[str], train_epochs: int) -> dict:
-    if "warmup_epochs" not in SEARCH_SPACE and "warmup_epochs" in search_parameters:
```

Comment:
Why remove `warmup_epochs" not in SEARCH_SPACE`? If one uses a self-defined search space, it would rewrite the search range to `tune.quniform(lower=1, upper=train_epochs // 2, q=1)`.

Reply:
Good catch. We will rewrite it.
We removed it as initially it was always returning "None" (which was its initialization for default search space dict) for each case.